### PR TITLE
Conglei graph refine

### PR DESCRIPTION
**Pull request by conglei**
In this PR, The layout algorithm is refined by better scaling the size of graph to fit the canvas size.

### CODE REVIEW

File: packages/network/src/chart/Network.jsx [hunk 1]
```diff
@@ -73,16 +86,64 @@ class Network extends React.PureComponent {
...
+    const dataYRange = range.y.max - range.y.min;
+
+    const zoomLevel = Math.max(
+      dataXRange / actualWidth,
+      dataYRange / actualheight,
+    );
+
+    function xScale(x) {
+      const offset = margin.left +
+      (((actualWidth - (dataXRange / zoomLevel)) / 2) - (range.x.min / zoomLevel));
```
Comment (williaster) [inline packages/network/src/chart/Network.jsx]: you could potentially separate these calculations out into separate variables with meaningful names to make this more readable

Comment (williaster) [reply packages/network/src/chart/Network.jsx]: also, can you move (x and y) `offset`s out of the functions so you only have to compute them once?

**Comment by conglei** [comment]
Done with the revison :) thanks for the comments. And I clean the code a bit. and 

1): Adding responsive support.

2): Show "waiting' words before the graph is layed out. (since the graph itself may not have x and y props, and the layout algorithm will add x and y for each node).

**Comment by williaster** [comment]
Great, lgtm! 🏆 

I may make the label optional before cutting a new release. I'm also not quite sure about best practice for `timeOut`s in tests, I think it can make them less reliable because simple race conditions can make them fail 🤔

**Pull request merged**